sdk/jobs/automl-standalone-jobs/ONNX_batch_model_generator_automl_for_images.py

```diff
@@ -0,0 +1,357 @@
+        onnx_model = onnx.load(onnx_model_file_path)
+        onnx.checker.check_model(onnx_model)
+        logger.info(
+            "ONNX model generation for the batch size {} is successful".format(
+                batch_size
+            )
+        )
+    except Exception as e:
+        logger.error("ONNX model generation or Schema validation error: " +
+                     str(e))
+        raise
+
+
+if __name__ == "__main__":
+
+    parser = argparse.ArgumentParser()
+    # add arguments
+    parser.add_argument(
+        "--model_name",
+        type=str,
+        required=True,
+        help="Model name supported from yolov5, fasterrcnn_resnet152_fpn,\
+                        maskrcnn_resnet101_fpn etc.",
+    )
+    parser.add_argument(
+        "--batch_size", type=int, required=True,
+        help="No. of samples per batch"
+    )
+    parser.add_argument(
+        "--height_onnx",
+        type=int,
+        required=True,
+        help="Height of the image for ONNX model",
+    )
+    parser.add_argument(
+        "--width_onnx",
+        type=int,
+        required=True,
+        help="Width of the image for ONNX model",
+    )
+    parser.add_argument(
+        "--job_name", type=str, required=True, help="job name of the run"
+    )
+    parser.add_argument(
+        "--task_type", type=str, required=True,
+        help="Task type in automl for images"
+    )
+    parser.add_argument(
+        "--box_score_thresh",
+        type=float,
+        required=True,
+        help="During inference, only return proposals with a \
+              classification score greater than box_score_thresh",
+    )
+
+    parser.add_argument(
+        "--min_size",
+        type=int,
+        required=False,
+        help="Minimum size of the image to be rescaled before feeding it to \
+        the backbone",
+    )
+    parser.add_argument(
+        "--max_size",
+        type=int,
+        required=False,
+        help="Maximum size of the image to be rescaled before feeding it to \
+        the backbone",
+    )
+    parser.add_argument(
+        "--box_nms_thresh",
+        type=float,
+        required=False,
+        help="Non-maximum suppression (NMS) threshold for the prediction head",
+    )
+    parser.add_argument(
+        "--box_detections_per_img",
+        type=int,
+        required=False,
+        help="Maximum number of detections per image, for all classes",
+    )
+    parser.add_argument(
+        "--img_size",
+        type=int,
+        required=False,
+        help="Image size for train and validation",
+    )
+    parser.add_argument(
+        "--model_size",
+        type=str,
+        required=False,
+        help="Size of the model from small, medium, large, xlarge",
+    )
+    parser.add_argument(
+        "--box_iou_thresh", type=float, required=False, help="IoU threshold"
+    )
+
+    # parse arguments
+    args = parser.parse_args()
+    experiment = Run.get_context().experiment
```

Comment:
Why is this changed to using "Run.get_context()" ?
To get the mlflow_tracking_uri, we can create a MLClient using subscription_id, resource_group and workspace.
and get it using

MLFLOW_TRACKING_URI = ml_client.workspaces.get(
    name=ml_client.workspace_name
).mlflow_tracking_uri

Reply:
if I don't get it from context, then scriptrunconfig needs to be authenticated manually each time. One customer reported that as well on ONNX. That's why I modified to get_context. I have seen this methodology across lot of scripts in this repo ex: https://github.com/Azure/azureml-examples/blob/a9c5ff547feb2d3c266313da9750b0f0cd0c477c/cli/jobs/pipelines-with-components/rai_pipeline_adult_analyse/components/src_train_logreg/train.py